src/js/components/Table/Table.js

```diff
@@ -47,6 +47,14 @@ class Table extends PureComponent {
     this.setState({ ...tableState })
   }
 
+  componentDidUpdate(prevProps) {
+    const { rows } = this.props
+
+    if (prevProps.rows.length !== rows.length) {
```

Comment:
Could there be an issue here with the table not updating correctly?  Just wonder if the filtered results might end up having the same length, what would happen?  Am I overthinking here?  I'm not sure if we would want it to "always" update though.  @Ollo you have any thoughts on this?

Reply:
I'm also curious if we want a componentWillReceiveProps instead?

Reply:
Hmm.. This feels a bit like a 🔨to me as a prop or state change should automatically trigger a re-render. Have we tested converting this from a `PureComponent` to see if we are getting bitten by the shallow compare?

Reply:
Not sure what the hammer means... 

The component is only updating state when the component mounts, which is only one time. It should happen when it updates as well as long as something has changed.

Good call about testing out converting from PureComponent. I will try that. Perhaps that's also the reason why sorting is not working properly.

I need the table to update when there are changes, otherwise how can I display filtered items?

I really hope that as this codebase grows we can introduce tests.

Reply:
![image](https://user-images.githubusercontent.com/8335640/40946473-44961964-6823-11e8-83d7-0bc159c1fdba.png)

Reply:
Here, the table is only passing 1 row prop, which is correct because I am filtering on `Non Player`. `state.items`, which are the row items, are still holding the original 13 objects that it initialized with.